### ISSUE TITLE
[css-conditional] Nested at-rules within at-rule()

### ISSUE THREAD

**Issue by andruud** [w3c/csswg-drafts#11117]
Some at-rules are only valid within a certain _other_ at-rule, e.g. margin rules within [@page](https://drafts.csswg.org/css-page/#syntax-page-selector), or `@swash` (etc) within [@font-feature-values](https://drafts.csswg.org/css-fonts/#font-feature-values-syntax).

So, how do these evaluate?

```
@supports at-rule(@swash) {}
@supports at-rule(@top-left) {}
```

It seems unwise to just _ignore_ context, since a given `@foo` may be specified to exist in more than one context, with different meanings (and possibly different descriptors being valid).

This was discussed during https://github.com/w3c/csswg-drafts/issues/2463, but I don't see a conclusion.

**Comment by bramus** [comment]
In https://github.com/w3c/csswg-drafts/issues/6966#issuecomment-2339051165 I suggested to allow authors to either test for an at-rule’s name or an actual full at-rule.

That would allow the following:

```css
@supports at-rule(@page { @top-left { ... }}) {
  …
}
```

… which I believe fixes this issue.

**Comment by cdoublev** [comment]
`@supports ( <at-rule-name>#? , <declaration> ) { ... }`? Eg. `@supports (@page, @top-left, size: 1px)`.

I have not read all related discussions so I do not know the motivations for using `at-rule`, and that might not always be enough to fully define the context. But mirroring `CSS.supports()` seems nice.

**Comment by bramus** [comment]
> I have not read all related discussions so I do not know the motivations for using `at-rule`, and that might not always be enough to fully define the context. But mirroring `CSS.supports()` seems nice.

- Motivation: Feature detect support for certain at-rules. See https://brm.us/at-rule for an intro.

- Mirroring `CSS.supports`: That is already the css. Doing `@supports at-rule(@starting-style)` in CSS is equivalent to `CSS.supports('at-rule(@starting-style)')` in JS — this is similar to `selector()` feature detection.

**Comment by cdoublev** [comment]
Exact, I had `CSS.supports(<declaration>)` in mind whereas it takes `<supports-condition>`.

I already read your article before and forget about `selector()` and `font-tech()`, so now `at-rule()` makes sense. 

Single argument blocks/functions is probably easier for authors, even if it means feeding a whole rule to define a nesting context.

**Comment by SebastianZ** [comment]
> In [#6966 (comment)](https://github.com/w3c/csswg-drafts/issues/6966#issuecomment-2339051165) I suggested to allow authors to either test for an at-rule’s name or an actual full at-rule.

For what it's worth, allowing to test a full at-rule's syntax is also [what I suggested earlier](https://github.com/w3c/csswg-drafts/issues/5929#issuecomment-774475760) and [mentioned again in the discussion in #2463](https://github.com/w3c/csswg-drafts/issues/2463#issuecomment-813025456).
This allows testing nested rules as well as at-rules only consisting of preludes.

The main counter-argument back then was that it's verbose. Though like @bramus I believe it makes sense to allow both testing for the full rule _and_ offer a simpler way. Allowing to test for the full syntax ensures that every feature of at-rules _can_ be tested for. _In addition_ to that we can provide short forms to test them.

Sebastian

**Comment by tabatkins** [comment]
Yeah, I think allowing both the simple "just the name" and the full-parsing "the entire rule" possibilities is probably the way to go. The latter is verbose, but for most at-rules you can leave out quite a lot and still get something valid. It also neatly answers *any other* structural questions we might have about testing and relationships between at-rules.

**Comment by kbabbitt** [comment]
Tagging for discussion at the upcoming F2F.

Proposal: Context is not considered in the "just the name" form but is considered in the "entire rule" form.

Details in the explainer:
https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/main/AtRuleFeatureDetection/explainer.md#proposed-approach
